processing/src/main/java/org/apache/druid/math/expr/ExprEval.java

```diff
@@ -771,7 +789,7 @@ private static class DoubleExprEval extends NumericExprEval
 
     private DoubleExprEval(@Nullable Number value)
     {
-      super(value == null ? NullHandling.defaultDoubleValue() : (Double) value.doubleValue());
+      super(value == null ? null : value.doubleValue());
```

Comment:
Does the `.doubleValue()` add anything here?  It looks like it's storing a reference to a Number anyway?  I guess it's a forced cast?

Reply:
not really, can be simplified

Reply:
ah i was wrong, at least a lot of tests were counting on forced casts, i see lots of 
```
Expected :java.lang.Long<1>
Actual   :java.lang.Integer<1>
```
after removing it